internal/ls/types.go

```diff
@@ -15,3 +15,36 @@ type Location struct {
 	FileName string
 	Range    core.TextRange
 }
+
+type JsxAttributeCompletionStyle string
+
+const (
+	JsxAttributeCompletionStyleAuto   JsxAttributeCompletionStyle = "auto"
+	JsxAttributeCompletionStyleBraces JsxAttributeCompletionStyle = "braces"
+	JsxAttributeCompletionStyleNone   JsxAttributeCompletionStyle = "none"
+)
+
+type UserPreferences struct {
```

Comment:
Long term it'll be interesting to see if we do this the same way; LSP has a different idea of how preferences are communicated than what tsserver did. But for internal stuff, probably okay.